dv/uvm/icache/dv/ibex_icache_mem_agent/README.md

```diff
@@ -0,0 +1,22 @@
+# ICache Memory UVM Agent
+
+The ICache memory UVM agent models the instruction memory bus downstream of the ICache.
+This also includes the PMP machinery.
+
+The basic idea is that instruction memory is modelled with a single 32-bit *seed*.
+A simple hash function takes this seed and calculates 32 bits of memory for each (aligned) base address.
+The seed also determines a memory region which should result in PMP errors and another memory region that should result in memory errors.
+
+The agent exposes a slave interface to the core using the usual UVM architecture for a reactive slave, but it looks a little complicated because we need to spot PMP errors immediately (rather than on the next clock edge).
+The entire dance is as follows:
+
+  1. The monitor spots a new request (either because of a posedge on the `req` line or a change in the `addr` line).
+  1. It generates an `ibex_icache_mem_req_item` with `is_grant` field false and writes it to the sequencer's request port.
```

Comment:
Are these all meant to be `1.`?

Reply:
> Are these all meant to be `1.`?

@udinator .md does the right thing here, and is the easiest
way to maintain a numbered list so you can add / remove
easily.

Reply:
Ah okay, thanks for letting me know; I'll resolve this conversation.